### ISSUE TITLE
Test_ConnectToDeadNSEUsecase is unstable

### ISSUE THREAD

**Issue by Bolodya1997** [networkservicemesh/sdk#844]
## Failure Logs
https://github.com/networkservicemesh/sdk/runs/2377941534?check_suite_focus=true#step:5:75
```
--- FAIL: TestNsmgr (12.32s)
    --- FAIL: TestNsmgr/Test_ConnectToDeadNSEUsecase (5.00s)
        nsmgr_test.go:318: 
            	Error Trace:	nsmgr_test.go:318
            	Error:      	Received unexpected error:
            	            	rpc error: code = DeadlineExceeded desc = context deadline exceeded
            	            	Error returned from api/pkg/api/registry/networkServiceEndpointRegistryClient.Unregister
            	            	github.com/networkservicemesh/sdk/pkg/registry/core/trace.(*traceNetworkServiceEndpointRegistryClient).Unregister
            	            		/home/runner/work/sdk/sdk/pkg/registry/core/trace/nse_registry.go:120
            	            	github.com/networkservicemesh/sdk/pkg/registry/core/next.(*nextNetworkServiceEndpointRegistryClient).Unregister
            	            		/home/runner/work/sdk/sdk/pkg/registry/core/next/nse_registry_client.go:75
            	            	github.com/networkservicemesh/sdk/pkg/registry/common/sendfd.(*sendFDNSEClient).Unregister
            	            		/home/runner/work/sdk/sdk/pkg/registry/common/sendfd/client_linux.go:85
            	            	github.com/networkservicemesh/sdk/pkg/registry/core/trace.(*traceNetworkServiceEndpointRegistryClient).Unregister
            	            		/home/runner/work/sdk/sdk/pkg/registry/core/trace/nse_registry.go:117
            	            	github.com/networkservicemesh/sdk/pkg/registry/core/next.(*nextNetworkServiceEndpointRegistryClient).Unregister
            	            		/home/runner/work/sdk/sdk/pkg/registry/core/next/nse_registry_client.go:73
            	            	github.com/networkservicemesh/sdk/pkg/registry/common/refresh.(*refreshNSEClient).Unregister
            	            		/home/runner/work/sdk/sdk/pkg/registry/common/refresh/nse_registry_client.go:144
            	            	github.com/networkservicemesh/sdk/pkg/registry/core/trace.(*traceNetworkServiceEndpointRegistryClient).Unregister
            	            		/home/runner/work/sdk/sdk/pkg/registry/core/trace/nse_registry.go:117
            	            	github.com/networkservicemesh/sdk/pkg/registry/core/next.(*nextNetworkServiceEndpointRegistryClient).Unregister
            	            		/home/runner/work/sdk/sdk/pkg/registry/core/next/nse_registry_client.go:73
            	            	github.com/networkservicemesh/sdk/pkg/registry/common/serialize.(*serializeNSEClient).Unregister.func1
            	            		/home/runner/work/sdk/sdk/pkg/registry/common/serialize/nse_client.go:55
            	            	github.com/networkservicemesh/sdk/pkg/tools/multiexecutor.(*MultiExecutor).AsyncExec.func2.1
            	            		/home/runner/work/sdk/sdk/pkg/tools/multiexecutor/multi_executor.go:54
            	            	github.com/edwarnicke/serialize.(*Executor).process
            	            		/home/runner/go/pkg/mod/github.com/edwarnicke/serialize@v1.0.7/serialize.go:68
            	            	runtime.goexit
            	            		/opt/hostedtoolcache/go/1.16.3/x64/src/runtime/asm_amd64.s:1371
            	Test:       	TestNsmgr/Test_ConnectToDeadNSEUsecase
    --- FAIL: TestNsmgr/Test_ShouldCleanAllClientAndEndpointGoroutines (0.19s)
        nsmgr_test.go:682: 
            	Error Trace:	nsmgr_test.go:682
            	            				nsmgr_test.go:480
            	Error:      	Condition never satisfied
            	Test:       	TestNsmgr/Test_ShouldCleanAllClientAndEndpointGoroutines
Error:     leaks.go:78: found unexpected goroutines:
```

**Comment by edwarnicke** [comment]
@Bolodya1997 Is this still unstable?

**Comment by Bolodya1997** [comment]
@edwarnicke Nope, fixed as a part of #870.
